source/development/feature-flags.html.md

```diff
@@ -0,0 +1,76 @@
+
+### Extended release
+Where we release a new feature or change but not have it available by default. Allows users to implement a large or complex feature incrementally whilst enabling them to continue staying up to date with new releases of our library. Use an extended release to facilitate the rollout of a long term, wide reaching feature.
+
+### Extended deprecation
+A “reverse” extended release where we remove something but keep the feature available via a feature flag. This lets users continue using features we want to remove for an extended period of time, allowing them to move away from said features incrementally. **Only use an extended deprecation as a last resort**. Whilst extended releases enable us to roll out changes slowly, extended deprecations moreso than extended releases facilitate potential bad or outdated practice hanging around. We should feel empowered to remove things that are bad for users.
+
+## Additional considerations
+Once you’ve established that a feature flag is a useful option for the thing you’re working on, consider the following principles when implementing and operating your flag.
+
+### Try to minimise the scope of the change
+Try to have your flag controlling as little as possible. Minimising how much is behind a feature flag reduces the burden of technical management and the impact of the change itself. Check if there’s anything in your change that can be released outside the feature flag as a separate change.
+
+### Be aware of the need for multiple feature flags
+Your feature may impact multiple parts of our tech stack eg: the Sass and the javascript. In which case you should ensure that your flag is built in such a way that the impacted element states are “synced” where possible. For example, if you have a flag in a component macro that controls the markup of that macro but also requires a Sass change, your flag should automatically add a CSS class that applies that styling change.
+
+## The lifecycle of a feature flag
+Follow these instructions for applying, working with and finally getting rid of feature flags.
+
+### Implementing a feature flag
+Feature flags and the change they’re controlling should be released in a minor release aka: X.X.X. As part of this release, automated tests should be written to account for govuk-frontend with the feature flag on and off to check for unforeseen impact to parts of the codebase that the flag wasn’t intended to affect.
+
+At this stage, we should create github issues to remove the feature flag in a future major release and apply them to ourplanning accordingly.
+
+We communicate in release notes and guidance what this feature flag impacts, how to use it and how “experimental” the new change is.
+
+### Working with a feature flag in flight
+By turning a feature flag on, we presume a user is opting into the following conditions:
```

Comment:
```suggestion
By turning a feature flag on, you presume a user is opting into the following conditions:
```

Reply:
Should the sentences following this be in bullet points or should there be a summary list of bullet points related to the following sections/subheads?

Reply:
It should be. I think I planned for it to be a list and then forgot to mark it up. Will sort.